rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java

```diff
@@ -226,7 +251,13 @@ public static Completable inTransactionCompletable(SQLClient client, Function<SQ
    */
   public static Completable usingConnectionCompletable(SQLClient client, Function<SQLConnection, Completable> sourceSupplier) {
     return client.rxGetConnection().flatMapCompletable(conn -> {
-      return sourceSupplier.apply(conn).doFinally(conn::close);
+      try {
+        return sourceSupplier.apply(conn).doFinally(conn::close);
+      } catch (Throwable t) {
+        Exceptions.throwIfFatal(t);
```

Comment:
don't we want to throw after conn.close() ?

Reply:
That's a general practice you can find in the RxJava codebase: if exception is fatal (e.g. OOM) you shouldn't do anything anyway (except logging it).

Reply:
ok, I was not aware of this

Reply:
I mean the RxJava practice :-)